src/main/java/com/iexec/core/replicate/ReplicateSupplyService.java

```diff
@@ -112,15 +112,9 @@ public ReplicateSupplyService(ReplicatesService replicatesService,
 
         // filter the Tasks that have reached the contribution deadline
         List<Task> validTasks = runningTasks.stream()
-                .filter(task -> {
-                    if (task.isContributionDeadlineReached()) {
-                        contributionTimeoutTaskDetector.detect();
-                        return false;
-                    } else {
-                        return true;
-                    }
-                })
+                .filter(task -> ! task.isContributionDeadlineReached())
                 .collect(Collectors.toCollection(ArrayList::new));
+        if (validTasks.size() != runningTasks.size()) contributionTimeoutTaskDetector.detect();
```

Comment:
It was agreed that we don't use one-liners for readability purposes.

Reply:
fixed, new commit pushed